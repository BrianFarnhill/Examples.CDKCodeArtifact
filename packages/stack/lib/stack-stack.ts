import * as cdk from '@aws-cdk/core';
import * as brfarn from '@brfarn/construct'

export class StackStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here

    new brfarn.Construct(this, "TestConstruct", {})
  }
}
