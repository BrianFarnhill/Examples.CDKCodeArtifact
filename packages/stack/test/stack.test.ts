import { expect as expectCDK, matchTemplate, MatchStyle, SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as Stack from '../lib/stack-stack';

test('Empty Stack', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new Stack.StackStack(app, 'MyTestStack');
    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
});
