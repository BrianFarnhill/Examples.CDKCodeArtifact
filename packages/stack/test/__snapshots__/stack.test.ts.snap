// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Stack 1`] = `
Object {
  "Resources": Object {
    "TestConstructConstructQueue35DCEBD2": Object {
      "Properties": Object {
        "VisibilityTimeout": 300,
      },
      "Type": "AWS::SQS::Queue",
    },
    "TestConstructConstructQueueMyTestStackTestConstructConstructTopicC756E8756B81E37A": Object {
      "Properties": Object {
        "Endpoint": Object {
          "Fn::GetAtt": Array [
            "TestConstructConstructQueue35DCEBD2",
            "Arn",
          ],
        },
        "Protocol": "sqs",
        "TopicArn": Object {
          "Ref": "TestConstructConstructTopicE115B9F1",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "TestConstructConstructQueuePolicyA6EDD2D0": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sqs:SendMessage",
              "Condition": Object {
                "ArnEquals": Object {
                  "aws:SourceArn": Object {
                    "Ref": "TestConstructConstructTopicE115B9F1",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": Object {
                "Service": "sns.amazonaws.com",
              },
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "TestConstructConstructQueue35DCEBD2",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Queues": Array [
          Object {
            "Ref": "TestConstructConstructQueue35DCEBD2",
          },
        ],
      },
      "Type": "AWS::SQS::QueuePolicy",
    },
    "TestConstructConstructTopicE115B9F1": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;
